### ISSUE TITLE
aws resource synchronization problem

### ISSUE THREAD

**Issue by songant** [cloudquery/cloudquery#4627]
Error synchronizing aws resources using cloudquery, the configuration is as follows:

**aws.yml**

```yml
kind: source
spec:

  name: "aws"
  path: "cloudquery/aws"

  version: "v5.1.1" # latest version of aws plugin

  destinations: ["postgresql"]

  spec:
    regions:
      - us-east-1
    accounts:
      - local_profile: "/Users/shareit/.aws/credentials"
        id: "XXX"
    aws_debug: true
```

 **/Users/shareit/.aws/credentials**

```ini
[default]
aws_access_key_id = xxxxx
aws_secret_access_key = xxxxx
```

**Execution error**
```
Loading spec(s) from aws.yml, postgresql.yml
Starting migration for: aws -> [postgresql]
Migration completed successfully.
Starting sync for:  aws -> [postgresql]
Error: failed to sync source aws: failed to fetch resources from stream: rpc error: code = DeadlineExceeded desc = failed to sync resources: failed to create execution client for source plugin aws: error retrieving credentials: failed to refresh cached credentials, no EC2 IMDS role found, operation error ec2imds: GetMetadata, canceled, context deadline exceeded
```
thanks

**Comment by songant** [comment]
error

**Comment by erezrokah** [comment]
Hi @songant and thank you for opening the issue.

Can you trying removing `local_profile` from your configuration file, that is try the following:
```yml
kind: source
spec:

  name: "aws"
  path: "cloudquery/aws"

  version: "v5.1.1" # latest version of aws plugin

  destinations: ["postgresql"]

  spec:
    regions:
      - us-east-1
    accounts:
      - id: "XXX"
    aws_debug: true
```

I believe that should work. In general `local_profile` should point to the name of a profile inside the AWS credentials file (and not to the file itself). For example, with the following file

```ini
[default]
aws_access_key_id=xxxx
aws_secret_access_key=xxxx

[user1]
aws_access_key_id=xxxx
aws_secret_access_key=xxxx
```

`local_profile` should be either `default` or `user1`.

More about this in https://github.com/cloudquery/cloudquery/blob/58112854fb686653d33a0db9eed504a1feb13dee/plugins/source/aws/docs/configuration.md#accounts.

I'll also look into improving that error message/our docs

**Comment by songant** [comment]
thanks，I have solved my problem。correctly configured：

aws.yml
`
kind: source
spec:
  name: "aws"
  path: "cloudquery/aws"
  version: "v5.1.1" # latest version of aws plugin


  destinations: ["postgresql"]

  spec:
    regions:
      - us-east-1
    accounts:
      - id: "XXXX"
    aws_debug: true
`

The aws credentials file is in the default path.

**Comment by erezrokah** [comment]
Hi @songant, thank you for confirming the fix and sharing additional logs. Can you try deleting the existing log file `cloudquery.log` and do a fresh run? We append to the log so I would like to confirm the errors you're experiencing are indeed from the new run

**Comment by songant** [comment]
I deleted the log file, and then re-executed the synchronization command, the error is as follows:

`
Loading spec(s) from aws.yml, postgresql.yml
Starting migration for: aws -> [postgresql]
Migration completed successfully.
Starting sync for:  aws -> [postgresql]
Sync completed successfully. Resources: 34179, Errors: 1177, Panics: 0, Time: 5m43s
`


`
2022-11-16T08:57:29Z ERR table resolver finished with error error="operation error RAM: ListPrincipals, 1 validation error(s) found.\n- missing required field, ListPrincipalsInput.ResourceOwner.\n" client=XXXXXXXX:us-east-1 module=aws-src table=aws_ram_principals
2022-11-16T08:57:29Z INF fetch table finished client=848318613114:us-east-1 errors=1 module=aws-src resources=0 table=aws_ram_principals
`


`
2022-11-16T08:57:32Z ERR table resolver finished with error error="operation error KMS: ListAliases, https response error StatusCode: 400, RequestID: 1e697be6-16e6-4d19-ba9c-6068c9f3b637, api error AccessDeniedException: User: arn:aws:iam::XXXXXX:user/SGT-Tag-Service is not authorized to perform: kms:ListAliases on resource: * because no identity-based policy allows the kms:ListAliases action" client=XXXXXXXXX:us-east-1 module=aws-src table=aws_kms_aliases
`

The above is the main error message.

**Comment by erezrokah** [comment]
Thank you @songant, that's very helpful. I think the sync was mostly successful.
Let me look at those and get back to you. They seem like 2 different issues

**Comment by erezrokah** [comment]
Hi @songant the `missing required field, ListPrincipalsInput.ResourceOwner` error should have been fixed in [`v7.0.0` of the AWS plugin](https://github.com/cloudquery/cloudquery/releases/tag/plugins-source-aws-v7.0.0), released 12 minutes ago :). Do you mind trying with that version?

I'll keep looking at the KMS one

**Comment by erezrokah** [comment]
Hi @songant, I was able to sync the KMS aliases resourced with getting a permission error, so it seems the AWS credentials you're using don't have access to `kms:ListAliases`. You can read more about this here https://docs.aws.amazon.com/kms/latest/developerguide/aws-managed-policies.html.

Per my comments I'll resolve the issue, but please feel free to ask any further questions and I'll re-open it
